packages/@lwc/rollup-plugin/src/index.ts

```diff
@@ -120,17 +154,24 @@ module.exports = function rollupLwcCompiler(pluginOptions = {}) {
             // Extract module name and namespace from file path
             const [namespace, name] = path.dirname(id).split(path.sep).slice(-2);
 
-            const { code, map } = compiler.transformSync(src, id, {
+            const { code, map } = transformSync(src, id, {
                 name,
                 namespace,
-                outputConfig: { sourcemap: pluginOptions.sourcemap },
-                stylesheetConfig: pluginOptions.stylesheetConfig,
-                experimentalDynamicComponent: pluginOptions.experimentalDynamicComponent,
-                preserveHtmlComments: pluginOptions.preserveHtmlComments,
+                outputConfig: { sourcemap },
+                stylesheetConfig,
+                experimentalDynamicComponent: (pluginOptions as any).experimentalDynamicComponent,
+                preserveHtmlComments,
                 scopedStyles: scoped,
             });
 
-            return { code, map };
+            const rollupMap = map as SourceMapInput;
+            return { code, map: rollupMap };
         },
     };
-};
+}
+
+// In order to keep backward compatibility with commonjs format, we can't use the export default.
+// Using export default will result in consumer importing the plugin via
+// `require("@lwc/rollup-plugin").default`. We should revisit this for the next major release.
+module.exports = lwc;
+module.exports.default = lwc;
```

Comment:
It looks like this achieves the same effect:

```js
function lwc() { /* ... */}
export = lwc;
```

Although that would require removing `export interface` above, and I'm not sure of the impact of that.

In either case, I'm not sure we should add `module.exports.default = lwc;`. If we add a new way to import this package (e.g. `require('@lwc/rollup-plugin').default`), then wouldn't this potentially give us additional backwards-compat problems in the future, if people actually use it?

Reply:
Would simply removing `module.exports.default = lwc;` work? 
@pmdartus is your original intention here mean that if some consumer did `require('@lwc/rollup-plugin').default` then it won't work?

Reply:
> @pmdartus is your original intention here mean that if some consumer did `require('@lwc/rollup-plugin').default` then it won't work?

No. What I want to make sure is that existing rollup configs don't break. Rollup handles CSJ require and ESM imports in `rollup.config.js`. I would like to make sure that the plugin can still be imported via CommonJS. 

```js
const lwc = require('@lwc/rollup-plugin');
export default {
    input: './src/main.js',
    plugins: [lwc()],
};
```

I don't know if @nolanlawson's proposal solves this. It would require checking this locally.

Reply:
I've tested locally that having `module.exports = lwc;` will work when other projects use `const lwc = require('@lwc/rollup-plugin');` so I think this should work.

Reply:
> Would simply removing `module.exports.default = lwc;` work?

Yes, I think then the API surface would be exactly the same as before.